### PR TITLE
add e2e case for alert forward from managed clusters.

### PR DESCRIPTION
**Pull request by morvencao**
Add e2e case for alert forward from openshift clusters, by checking the `Watchdog` alert are forwarded to the hub, because `Watchdog` alert is always triggered, it is used to make sure the alert trigger is working. 

For *KS, add `Watchdog` alertrule to make sure the entire alerting pipeline is functional.

Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

File: tests/pkg/tests/observability_alert_test.go [hunk 1]
```diff
@@ -237,6 +247,83 @@ var _ = Describe("Observability:", func() {
 		klog.V(3).Infof("Successfully deleted CM: thanos-ruler-custom-rules")
 	})
 
+	It("[P2][Sev2][Observability][Stable] Should have alert named Watchdog forwarded to alertmanager generated (alert/g0)", func() {
```
Comment (clyang82) [inline tests/pkg/tests/observability_alert_test.go:250]: Suggest to use [Integration] firstly. we can move it to stable once it is stable.

Comment (morvencao) [reply tests/pkg/tests/observability_alert_test.go]: done.

**Comment by morvencao** [comment]
@clyang82 another look? New commits cover the *KS.

File: tests/pkg/utils/mco_managedcluster.go [hunk 2]
```diff
@@ -71,3 +72,85 @@ func ListManagedClusters(opt TestOptions) ([]string, error) {
...
+		if labels != nil {
+			vendorStr := ""
+			if vendor, ok := labels["vendor"]; ok {
+				vendorStr = vendor.(string)
+			}
+			obsControllerStr := ""
+			if obsController, ok := labels["feature.open-cluster-management.io/addon-observability-controller"]; ok {
+				obsControllerStr = obsController.(string)
+			}
+			if (vendorStr == "GKE" || vendorStr == "EKS" || vendorStr == "AKS") && obsControllerStr == "available" {
```
Comment (clyang82) [inline tests/pkg/utils/mco_managedcluster.go:143]: we support all *KS. so here suggest to use `vendorStr != `OpenShift`

Comment (morvencao) [reply tests/pkg/utils/mco_managedcluster.go]: done

**Comment by morvencao** [comment]
I found that in e2e-kind testing env, the managedclusters only have `vendor` label, no other labels, so `ListManagedClusters` function with `obs-addon-controller == avavilable` will always return empty list, so I reverted the `obs-addon-controller` label check.

**Comment by clyang82** [comment]
> I found that in e2e-kind testing env, the managedclusters only have `vendor` label, no other labels, so `ListManagedClusters` function with `obs-addon-controller == avavilable` will always return empty list, so I reverted the `obs-addon-controller` label check.

@elgnay Could you share why there is no `feature.open-cluster-management.io/addon-observability-controller` label for KinD cluster? is it by design? Thanks.

**Comment by morvencao** [comment]
@clyang82 Maybe we need to use https://github.com/open-cluster-management/registration-operator release-2.4 instead of release 2.3, not sure if that's is a new feature in 2.4.

**Comment by morvencao** [comment]
After upgrading https://github.com/open-cluster-management/registration-operator to `release-2.4`, the managedcluster label still only has `vendor`:

```
$ kubectl get managedcluster --show-labels 
NAME            HUB ACCEPTED   MANAGED CLUSTER URLS   JOINED   AVAILABLE   AGE     LABELS
local-cluster   true           https://localhost      True     True        7m59s   vendor=GKE
```

**Comment by elgnay** [comment]
> After upgrading https://github.com/open-cluster-management/registration-operator to `release-2.4`, the managedcluster label still only has `vendor`:
> 
> ```
> $ kubectl get managedcluster --show-labels 
> NAME            HUB ACCEPTED   MANAGED CLUSTER URLS   JOINED   AVAILABLE   AGE     LABELS
> local-cluster   true           https://localhost      True     True        7m59s   vendor=GKE
> ```

Yes, it's a feature of 2.4 release. The addon labels are generated in my Kind env. Please make sure
1. the registration-hub-controller is running;
2. the addon CR is created in the cluster namespace;

**Pull request merged**